### PR TITLE
react-onclickoutside: return an AbstractComponent with instance

### PR DESCRIPTION
**Pull request by nifgraup**
Abstract component is new in 0.89.0,

Followed flow docs on HOCs [1] to declare the getInstance() method
react-onclickoutside provides [2] to access wrapped component instance.

Added two tests.

[1] https://flow.org/en/docs/react/hoc/#toc-preserving-the-instance-type-of-a-component
[2] https://github.com/Pomax/react-onclickoutside#but-how-can-i-access-my-component-it-has-an-api-that-i-rely-on

### CODE REVIEW

File: definitions/npm/react-onclickoutside_v6.x.x/flow_v0.89.0-/test_react-onclickoutside_v6.x.x.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import onClickOutside, { IGNORE_CLASS_NAME } from 'react-onclickoutside';
+
+IGNORE_CLASS_NAME.slice();
+
+// $ExpectError Cannot perform arithmetic operation because string literal `ignore-react-onclickoutside` [1] is not a number.
+IGNORE_CLASS_NAME - 1;
+
+// class and const are in same line to make all flow versions unhappy in the same line.
+// $ExpectError Cannot call `onClickOutside` because string [1] is incompatible with boolean [2].
+class FailComponent extends React.Component<{ preventDefault: string }> {render() {return <div/>;}} const FailComponentWithClickOutside = onClickOutside(FailComponent);
```
Comment (pascalduez) [inline definitions/npm/react-onclickoutside_v6.x.x/flow_v0.89.0-/test_react-onclickoutside_v6.x.x.js]: Could you format/prettify this?

**Comment by pascalduez** [comment]
Thanks!

Could you wrap/structure the tests in a few [`describe/it`](https://github.com/flow-typed/flow-typed/blob/master/CONTRIBUTING.md#use-describe-and-it-blocks-to-limit-scope) ?

File: definitions/npm/react-onclickoutside_v6.x.x/flow_v0.89.0-/test_react-onclickoutside_v6.x.x.js (hunk 1 above)
Comment (nifgraup) [reply definitions/npm/react-onclickoutside_v6.x.x/flow_v0.89.0-/test_react-onclickoutside_v6.x.x.js]: done

**Comment by nifgraup** [comment]
made requested fixes

**Pull request merged**